code/cycler/src/wattrex_battery_cycler/mid/mid_dabs/mid_dabs.py

```diff
@@ -0,0 +1,329 @@
+        Args:
+            current_ref (int): [current in mA]
+            limit_ref (int): [limit reference, for the epc could be mV/dW/ms the rest of devices
+                            is mV]
+        """
+        try:
+            if CyclerDataDeviceTypeE.EPC in self.device_type:
+                self.epc.set_cc_mode(current_ref, limit_type, limit_ref)
+            # elif CyclerDataDeviceTypeE.BISOURCE in self.device_type:
+            #     self.bisource.set_cc_mode(current_ref, limit_ref)
+            # elif (CyclerDataDeviceTypeE.SOURCE in self.device_type and
+            #     CyclerDataDeviceTypeE.LOAD in self.device_type):
+            #     if current_ref>0:
+            #         self.load.disable()
+            #         self.source.set_cc_mode(current_ref, limit_ref)
+            #     else:
+            #         # TODO: upgrade DrvRs to write limits when setting modes
+            #         self.source.disable()
+            #         self.load.set_cc_mode(current_ref)
+            else:
+                log.warning("This device is not able to change to CC modes.")
+        except Exception as err:
+            log.error(f"Error while setting cc mode: {err}")
+            raise Exception("Error while setting cc mode") from err #pylint: disable= broad-exception-raised
+
+    def set_cp_mode(self, pwr_ref: int, limit_type: CyclerDataPwrLimitE, limit_ref: int) -> None:
+        """Set the CP mode with the specified limits, only possible in the epc.
+
+        Args:
+            pwr_ref (int): [description]
+            limit_type (CyclerDataPwrLimitE): [description]
+            limit_ref (int): [description]
+        """
+        try:
+            if CyclerDataDeviceTypeE.EPC in self.device_type:
+                self.epc.set_cp_mode(pwr_ref, limit_type, limit_ref)
+            else:
+                log.warning('This device is incompatible with power control mode')
+        except Exception as err:
+            log.error(f"Error while setting cp mode: {err}")
+            raise Exception("Error while setting cp mode") from err #pylint: disable= broad-exception-raised
+
+    def set_wait_mode(self, time_ref: int = 0):
+        """Set the wait mode for the device.
+        To set the wait mode in epc must write argument time_ref = number_in_ms
+        """
+        try:
+            if CyclerDataDeviceTypeE.EPC in self.device_type:
+                self.epc.set_wait_mode(limit_ref = time_ref)
+            else:
+                self.disable()
+        except Exception as err:
+            log.error(f"Error while setting wait mode: {err}")
+            raise Exception("Error while setting wait mode") from err #pylint: disable= broad-exception-raised
+
+    def set_limits(self, ls_volt: tuple | None = None, ls_curr: tuple | None = None,
+                   ls_pwr: tuple | None = None, hs_volt: tuple | None = None,
+                   temp: tuple | None = None) -> None:
+        """Set the limits of the ECP.
```

Comment:
Set limits of the pwr device, not EPC

Reply:
Future versions, now only epc has limits